src/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -130,12 +130,33 @@ function getSpaceAvailable(storagePools, connectionName) {
     return space;
 }
 
+function getVmName(vmName, connectionName, vms, os) {
+    if (!isEmpty(vmName.trim()))
+        return vmName;
+
+    let retName = connectionName;
+    if (os)
+        retName += '-' + os.shortId;
+
+    const date = new Date();
+    retName += '-' + date.getFullYear() + "-" + (date.getMonth() + 1) + "-" + date.getDate();
+
+    let tmpRetName = retName;
+    // VM with same name already exists, append a character at the end, starting with 'B'
+    for (let i = 66; vms.some(vm => vm.name === tmpRetName) && i <= 91; i++) {
+        // Could not generate name which doesn't collide with any other VM name
+        if (i === 91)
+            return "";
+        tmpRetName = retName + '-' + String.fromCharCode(i);
+    }
+
+    return tmpRetName;
+}
+
 function validateParams(vmParams) {
     const validationFailed = {};
 
-    if (isEmpty(vmParams.vmName.trim()))
-        validationFailed.vmName = _("Name must not be empty");
```

Comment:
getVmName() can fail and return an empty string, so it seems we still actually need this?